core/common/src/main/java/alluxio/extensions/ExtensionFactory.java

```diff
@@ -13,33 +13,30 @@
 
 import alluxio.conf.AlluxioConfiguration;
 
-import javax.annotation.Nullable;
-
 /**
  * A factory class for creating instance of {@link T} based on configuration {@link S}.
  * @param <T> The type of instance to be created
  * @param <S> the type of configuration to be used when creating the extension
  */
-public interface ExtensionFactory<T, S> {
+public interface ExtensionFactory<T, S extends AlluxioConfiguration> {
```

Comment:
Will this break 3rd party ufses?

Reply:
b/w which versions? 
- starting 2.0-preview this does not break external ufses
- yes for older versions, since we did not have AlluxioConfiguration back then

Reply:
Okay, it would be good to notify the projects in https://github.com/Alluxio/alluxio-extensions